src/Interpreters/DDLTask.h

```diff
@@ -95,6 +96,9 @@ struct DDLLogEntry
     /// Only for DatabaseReplicated.
     std::optional<UUID> parent_table_uuid;
 
+    UUID initiator_user;
+    std::vector<UUID> initiator_user_roles;
```

Comment:
I suppose it's better to use names just because it's more general (and for ReplicatedAccessStorage it's almost the same). Also we need a way to enable/disable this feature, probably via some configuration setting, because not everyone will want it.

Reply:
This is disabled by default right now with the distributed_ddl_entry_format_version